utils/promiseTimeout.js

```diff
@@ -5,17 +5,19 @@
  * @param {Promise} promise Promise to timeout
  * @returns {Promise} Promise that rejects in <ms> milliseconds
  */
-const promiseTimeout = function(ms, promise) {
+export default function(ms, promise) {
+  let timerId
   // Create a promise that rejects in <ms> milliseconds
   let timeout = new Promise((resolve, reject) => {
-    let id = setTimeout(() => {
-      clearTimeout(id)
+    timerId = setTimeout(() => {
+      clearTimeout(timerId)
       reject(`Timed out.`)
     }, ms)
   })
 
   // Returns a race between our timeout and the passed in promise
-  return Promise.race([promise, timeout])
+  return Promise.race([promise, timeout]).then(value => {
+    timerId && clearTimeout(timerId)
```

Comment:
Should we also clear this in the event that the promise rejects? (in catch block)

Reply:
absolutely